### PR TITLE
Feat/student pointallocation

### PR DESCRIPTION
**Pull request by gioyap**
I completed the responsive UI design of student db of point allocation history, it contains dropdown of sort button, filters have a modal and view details button have a modal also with blurred bg. I also fixed the error from eslint

### CODE REVIEW

File: app/(student)/point-allocations/page.jsx [hunk 1]
```diff
@@ -1,7 +1,311 @@
...
+import {AiFillEye, AiOutlineCaretDown, AiOutlineCaretUp} from 'react-icons/ai'
+import {BsXLg} from 'react-icons/bs'
+import {MdArrowDownward, MdArrowUpward} from 'react-icons/md'
 
-const PointAllocations = () => {
-  return <div>PointAllocations</div>;
-};
+ const PointAllocation = () => {
+  const initialIsOpenState = Array(data.length).fill(false);
+  const [isOpen, setIsOpen] = useState(initialIsOpenState);
```
Comment (Just-a-NoobieDev) [inline app/(student)/point-allocations/page.jsx:12]: pwede mo naman na dito ipasa sa useState yung false mismo e haha pinahaba mo pa  code
```suggestion
  const [isOpen, setIsOpen] = useState(false);
```

File: app/(student)/point-allocations/page.jsx [hunk 2]
```diff
@@ -1,7 +1,311 @@
...
+  const [isUp, setIsUp] = useState(false);
+  const [isSort, setIsSort] = useState(false);
 
-export default PointAllocations;
+  const toggleTransaction = (index) => {
+    setIsOpen((prev) => {
+      const newState = [...prev];
+      newState[index] = !newState[index];
+      return newState;
+    });
```
Comment (Just-a-NoobieDev) [inline app/(student)/point-allocations/page.jsx:24]: pwede to pasimplehin like
```suggestion
  const toggleTransaction = (index) => {
    setIsOpen((prev) => !prev);
```

File: app/(student)/point-allocations/page.jsx [hunk 3]
```diff
@@ -1,7 +1,311 @@
...
-  return <div>PointAllocations</div>;
-};
+ const PointAllocation = () => {
+  const initialIsOpenState = Array(data.length).fill(false);
+  const [isOpen, setIsOpen] = useState(initialIsOpenState);
+  const [selectedTransaction, setSelectedTransaction] = useState(null);
+  const [isToggled, setIsToggled] = useState(false);
+  const [isClicked, setIsClicked] = useState(false);
+  const [isUp, setIsUp] = useState(false);
+  const [isSort, setIsSort] = useState(false);
```
Comment (Just-a-NoobieDev) [inline app/(student)/point-allocations/page.jsx:17]: @gioyap Palagay ng mga comment para kanino mga tong useState na to

File: app/(student)/point-allocations/page.jsx [hunk 4]
```diff
@@ -1,7 +1,311 @@
...
+                  className="flex gap-2 p-2 active:text-gray-400 text-xs md:text-sm lg:text-md"
+                  onClick={() => setIsToggled((prev) => !prev)}
+                >
+                  <AiFillEye />
+                  <p>View details</p>
+                </button>
+                </div>
+              ) : null}                 
+          </div>
+        </div>
```
Comment (Just-a-NoobieDev) [inline app/(student)/point-allocations/page.jsx:83]: @gioyap gawan mo ito sarili nga component kase reusable component to para mas madali maintain code nato

File: app/(student)/point-allocations/page.jsx [hunk 5]
```diff
@@ -1,7 +1,311 @@
...
+                  </div>
+                )}
+            </div>
+
+          <div className='col-span-2 xl:col-span-4 bg-white flex justify-end pt-4'> 
+              <button className='bg-[#5B801A] w-auto p-2 px-8 text-white font-bold text-xs md:text-sm lg:text-md xl:text-lg hover:bg-lime-800 active:bg-lime-900 '>Filter</button>
+          </div> 
+        </div>   
+      </div>
+      }
```
Comment (Just-a-NoobieDev) [inline app/(student)/point-allocations/page.jsx:134]: palagyan comment para saan tong block of code na to

File: app/(student)/point-allocations/page.jsx [hunk 6]
```diff
@@ -1,7 +1,311 @@
...
+                              Descending
+                              <MdArrowDownward />
+                            </div>
+                        </li>      
+                      </ul>
+                      <div className=' bg-white pt-8'> 
+                        <button className='bg-[#5B801A] w-auto py-1 px-6 text-white font-bold text-xs md:text-sm lg:text-sm xl:text-md hover:bg-lime-800 active:bg-lime-900 '>Apply</button>
+                      </div> 
+                    </div>
+      )}
```
Comment (Just-a-NoobieDev) [inline app/(student)/point-allocations/page.jsx:161]: same palagyan lang comment

File: app/(student)/point-allocations/page.jsx [hunk 7]
```diff
@@ -1,7 +1,311 @@
...
+                <div className=' col-span-1 md:col-span-1 pt-2 text-xs md:text-sm lg:text-md '><h3>Number of Bottles: </h3></div>
+                <div className=' col-span-2 md:col-span-3 bg-white p-4 border-2 border-gray-500 '></div>
+
+                <div className=' col-span-1 md:col-span-1 pt-2 text-xs md:text-sm lg:text-md '><h3>Transaction Date: </h3></div>
+                <div className=' col-span-2 md:col-span-3 bg-white p-4 border-2 border-gray-500 '></div>
+
+                <div className=' col-span-1 md:col-span-1 pt-2 text-xs md:text-sm lg:text-md '><h3>Converted Points: </h3></div> 
+                <div className=' col-span-2 md:col-span-3 bg-white p-4 border-2 border-gray-500 '></div>                                             
+            </div>
+          </div>
```
Comment (Just-a-NoobieDev) [inline app/(student)/point-allocations/page.jsx:183]: gawan mo din to ng component para maging reusable sa ibang page

**Review by Just-a-NoobieDev — CHANGES_REQUESTED** [review]
Pa resolve all review @gioyap before merge thank you

File: app/(student)/point-allocations/page.jsx (hunk 2 above)
Comment (gioyap) [reply app/(student)/point-allocations/page.jsx:24]: pre sinunod ko yan kaso biglang hindi nag function yung threedots sa table

File: app/(student)/point-allocations/page.jsx (hunk 1 above)
Comment (gioyap) [reply app/(student)/point-allocations/page.jsx:12]: same dito pre, somehow magkacorrelate sila ng line 19 to 24 haha. may lumalabas na error kapag pinipindot ko yung threedots.

File: app/(student)/point-allocations/page.jsx (hunk 2 above)
Comment (Just-a-NoobieDev) [reply app/(student)/point-allocations/page.jsx:24]: ano error

File: app/(student)/point-allocations/page.jsx (hunk 1 above)
Comment (Just-a-NoobieDev) [reply app/(student)/point-allocations/page.jsx:12]: ano nalabas na error?

**Review by Just-a-NoobieDev — APPROVED** [review]

**Pull request merged**